README.md

```diff
@@ -105,6 +105,57 @@ _Optional_ - Defaults to `true`
 Enable or disable the Reset Password Cleaner which handles expired reset password 
 requests that may have been left in persistence.
 
+## Advanced Usage
+
+### Purging `ResetPasswordRequest` objects from persistence
+
+The `ResetPasswordRequestRepositoryInterface::removeRequests()` method, which is 
+implemented in the 
+[ResetPasswordRequestRepositoryTrait](https://github.com/SymfonyCasts/reset-password-bundle/blob/main/src/Persistence/Repository/ResetPasswordRequestRepositoryTrait.php),
+can be used to remove all request objects from persistence for a single user. This 
+differs from the 
+[garbage collection mechanism](https://github.com/SymfonyCasts/reset-password-bundle/blob/df64d82cca2ee371da5e8c03c227457069ae663e/src/Persistence/Repository/ResetPasswordRequestRepositoryTrait.php#L73)
+which only removes _expired_ request objects for _all_ users automatically.
+
+Typically, you'd call this method when you need to remove request object(s) for 
+a user who changed their email address due to suspicious activity and potentially
+has valid request objects in persistence with their "old" compromised email address.
+
+This method relies on the user objects `primary key` being `immutable`.
+E.g. `User::id = UUID` not something like `User::id = 'john@example.com'`.
+
+```php
+// ProfileController
+
+#[Route(path: '/profile/{id}', name: 'app_update_profile', methods: ['GET', 'POST'])]
+public function profile(Request $request, User $user, ResetPasswordRequestRepositoryInterface $repository): Response
+{
+    $form = $this->createFormBuilder($user)
+        ->add('email', EmailType::class)
+        ->add('save', SubmitType::class, ['label' => 'Save Profile'])
+        ->getForm()
+    ;
+    
+    $form->handleRequest($request);
+    
+    if ($form->isSubmitted() && $form->isValid()) {
+        $newEmail = $form->get('email')->getData();
+        
+        if ($newEmail !== $user->getEmail()) {
+            // The user changed their email address.
+            // Remove any old reset requests for the user.
+            $repository->removeRequests($user);
+        }
+        
+        // Persist the user object...
+        
+        return $this->render('success.html.twig');
+    }
+    
+    return $this->render('profile.html.twig', ['form' => $form]);
+}
+```
```

Comment:
Should we convey that "hey, just because we're showing you how to flush requests because the email changed.. Doesnt mean you _need_ to flush requests everytime a user changes their email..."? I started to add something to that effect following the example, but I'm not sure if that will be confusing to someone who doesn't understand that this method is more for "security-ish" use cases..

Reply:
I think it's ok how it is now